lib/replay_api/request_deliverer.rb

```diff
@@ -0,0 +1,63 @@
+module ReplayApi
+  class RequestDeliverer
+    include Singleton
+
+    attr_writer :configuration
+    attr_writer :connection
+    attr_reader :request_buffer
+    attr_reader :time_last_write
+    attr_reader :delivery_thread
+
+    def initialize(configuration=ReplayApi.configuration)
+      @configuration ||= configuration
+      @time_last_write = Time.now
+      @request_buffer = Queue.new
+
+      start_delivery_thread
+    end
+
+    def post(request)
+      @request_buffer.push(request)
+    end
+
+    private
+
+    def start_delivery_thread
+      return if @delivery_thread
+
+      @delivery_thread = Thread.new do
+        loop do
+          requests = []
+          (requests << request_buffer.pop) until requests.length > 50 or time_since_last_write > 0.1
```

Comment:
Yeah, I would maybe make those numbers methods so it's clearer what they are and I would even move the logic to predicate methods.

```
def batch_full?(requests)
  requests.length > max_batch_size
end

def max_batch_size
  50
end

def max_delay_exeeded?
  (Time.now - time_last_write) > max_delay
end

def max_delay
  0.1
end
```